modules/porous_flow/include/materials/PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity.h

```diff
@@ -0,0 +1,72 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef POROUSFLOWTOTALGRAVITATIONALDENSITYFULLYSATURATEDFROMPOROSITY_H
+#define POROUSFLOWTOTALGRAVITATIONALDENSITYFULLYSATURATEDFROMPOROSITY_H
+
+#include "PorousFlowTotalGravitationalDensityBase.h"
+
+// Forward Declarations
+class PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity;
+
+template <>
+InputParameters validParams<PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity>();
+
+/**
+ * Material designed to provide the density of the porous medium for the
+ * fully-saturated case. Density is calculated as a
+ * weighted average of the fluid and solid densities:
+ * density = phi * rho_f + (1 - phi) * rho_s
+ * where phi is porosity, rho_f is fluid density and rho_s is solid
+ * density (assumed constant).
+ */
+class PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity
+    : public PorousFlowTotalGravitationalDensityBase
+{
+public:
+  PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity(const InputParameters & parameters);
+
+protected:
+  void initQpStatefulProperties() override;
+  void computeQpProperties() override;
+
+  /**
+   * Fluid density
+   * Over-ride in derived classes to implement other fluid density formulations
+   *
+   * @param ph phase index
+   * @return fluid density
+   */
+  virtual Real rho_f(unsigned int ph = 0) const;
+
+  /**
+   * Derivatives of fluid density wrt a variable
+   * Over-ride in derived classes to implement other fluid density formulations
+   *
+   * @param ph phase index
+   * @param v variable index
+   * @return derivatives of fluid density
+   */
+  virtual Real drho_f_dvar(unsigned int ph = 0, unsigned int v = 0) const;
```

Comment:
Remove the
```
unsigned int ph = 0
```
argument, and remove the default from v, ie just make it
```
unsigned int v
```
There's no need for the default.